### DISCUSSION TITLE
How to map a Boolean property with an Integer in Mapperly?

### DISCUSSION THREAD

**Discussion by naeemabbas05** [riok/mapperly#1156] [Q&A]
Currently we are using AutoMapper and we are migrating to Mapperly. I have the following scenario, 

```csharp
Class A{
    public int isActive {get; set; }
}

Class AA{
    public bool isActive { get; set: }
}


Public void AutoMapperConfig(){
            CreateMap<AA, A>()
                .ForMember(d => d.isActive, opt => opt.MapFrom(s => s.isActive != null && s.isActive != 0))
}
```
So I need to figure how could I handle the same case in Mapperly, I studied the documentation, but couldn't figure it out, anyone can help me out here?

**Comment by latonz** [comment]
Have you tried using a [user-implemented mapping](https://mapperly.riok.app/docs/configuration/user-implemented-methods/)?

**Reply by naeemabbas05** [reply]
@latonz I wrote a mapping method, the problem I am facing right now is, the code is throwing the following error. 

> Error	RMG007	Could not map member A.isActive of type int to AA.isActive of type bool	AA

Now, should I add MappingIgnore annotation? I guess it will not look for customer mapping method, if I wrote a mappingignore.

**Reply by latonz** [reply]
No, MapperIgnore* attribute is not needed if you want the property to be mapped.

**Reply by naeemabbas05** [reply]
```
[Mapper]
public partial class CarMapper
{
    private bool IntToBoolean(int? value) => value != 0;

    public partial CarDto CarToCarDto(Car car);

}
```
it keep throwing compile time error, "Error RMG007 Could not map member CarDto.isActive of type int to Car.isActive of type bool". How should I stop the compile time error.

**Reply by latonz** [reply]
Are your classes/mapper in a nullable context?

**Reply by naeemabbas05** [reply]
No, my class and properties are not nullable, Plus I don't understand you clearly. Do you want me to provide you more code?

**Reply by latonz** [reply]
Can you provide a minimal repro as a github repo?

**Reply by latonz** [reply]
Probably the parameter type of the single paramter of `IntToBoolean` should be `int` instead of `int?`?
